File: web_api/schemas/shop_onboarding.py
from pydantic import BaseModel, Field


class OnboardingCreateRequest(BaseModel):
    platform: str = "pdd"
    shop_name: str
    account_name: str
    password: str | None = Field(default=None, repr=False)
    operator: str = "local_admin"
    runner_mode: str = "fake"


class OnboardingSessionResponse(BaseModel):
    session_id: str
    platform: str
    shop_id: str | None = None
    shop_name: str | None = None
    safe_display: str = ""
    runner_mode: str = "fake"
    status: str
    step: str
    needs_sms_code: bool = False
    needs_captcha: bool = False
    captcha_image_ref: str | None = None
    error_summary: str | None = None
    created_at: str
    updated_at: str
    expires_at: str
    completed_at: str | None = None
    cancelled_at: str | None = None
    remote_browser_available: bool = False
    remote_browser_status: str | None = None
    vnc_url_ready: bool = False
    vnc_url: str | None = None
    real_shop_id_pending: bool = False


class SmsCodeSubmitRequest(BaseModel):
    sms_code: str = Field(min_length=1, max_length=12, repr=False)


class CaptchaSubmitRequest(BaseModel):
    captcha_code: str = Field(min_length=1, max_length=32, repr=False)


class CheckLoginResponse(OnboardingSessionResponse):
    pass


class AuthStatusResponse(BaseModel):
    shop_id: str
    platform: str
    account_name: str | None = None
    auth_status: str
    safe_display: str = ""
    last_login_at: str | None = None
    expires_at: str | None = None
    insecure_auth_storage: bool = False


class ChecklistItem(BaseModel):
    key: str
    label: str
    status: str
    required: bool
    summary: str


class OnboardingChecklistResponse(BaseModel):
    shop_id: str
    ready_for_ai: bool
    blocking_items: list[str]
    items: list[ChecklistItem]


class ValidationMarkPassedRequest(BaseModel):
    operator: str = "local_admin"
    summary: str = ""


class ValidationRunResponse(BaseModel):
    id: str
    shop_id: str
    status: str
    mode: str = "no_send"
    passed_count: int = 0
    failed_count: int = 0
    tested_at: str
    created_by: str
    summary: str = ""


class EnableAiRequest(BaseModel):
    operator: str = "local_admin"
    confirm: bool = False
    override: bool = False
    override_reason: str | None = None


class DisableAiRequest(BaseModel):
    operator: str = "local_admin"
    reason: str = "manual_disable"


class AiStatusResponse(BaseModel):
    shop_id: str
    ai_enabled: bool
    enabled_at: str | None = None
    enabled_by: str | None = None
    disabled_at: str | None = None
    disabled_by: str | None = None
    last_change_reason: str | None = None
    override_enabled: bool = False
    override_reason: str | None = None


class WorkerStatusResponse(BaseModel):
    shop_id: str
    status: str
    process_id: int | None = None
    last_seen_at: str | None = None
    websocket_status: str
    summary: str
    ai_enabled: bool = False
    consistency_status: str = "unknown"
    attention_required: bool = False
    recommended_action: str | None = None

File: web_frontend/src/api/shopOnboarding.ts
import { apiGet, apiPost } from "./client";

export type OnboardingSession = {
  session_id: string;
  platform: string;
  shop_id?: string | null;
  shop_name?: string | null;
  safe_display: string;
  runner_mode: "fake" | "real" | "remote_browser";
  status: string;
  step: string;
  needs_sms_code: boolean;
  needs_captcha: boolean;
  captcha_image_ref?: string | null;
  error_summary?: string | null;
  created_at: string;
  updated_at: string;
  expires_at: string;
  completed_at?: string | null;
  cancelled_at?: string | null;
  remote_browser_available?: boolean;
  remote_browser_status?: string | null;
  vnc_url_ready?: boolean;
  vnc_url?: string | null;
  real_shop_id_pending?: boolean;
  shop_identity_status?: string | null;
  auth_status?: string | null;
};

export type CreateOnboardingPayload = {
  platform: "pdd";
  shop_name: string;
  account_name: string;
  password?: string;
  operator?: string;
  runner_mode?: "fake" | "real" | "remote_browser";
};

export type BindShopIdentityPayload = {
  mall_id: string;
  shop_name?: string;
  operator?: string;
};

export type AuthStatus = {
  shop_id: string;
  platform: string;
  account_name?: string | null;
  auth_status: string;
  safe_display: string;
  last_login_at?: string | null;
  expires_at?: string | null;
  insecure_auth_storage: boolean;
};

export type ChecklistItem = {
  key: string;
  label: string;
  status: "passed" | "failed" | "warning" | "skipped" | string;
  required: boolean;
  summary: string;
};

export type OnboardingChecklist = {
  shop_id: string;
  ready_for_ai: boolean;
  blocking_items: string[];
  items: ChecklistItem[];
};

export type ValidationRun = {
  id: string;
  shop_id: string;
  status: string;
  mode: string;
  passed_count: number;
  failed_count: number;
  tested_at: string;
  created_by: string;
  summary: string;
};

export type AiStatus = {
  shop_id: string;
  ai_enabled: boolean;
  enabled_at?: string | null;
  enabled_by?: string | null;
  disabled_at?: string | null;
  disabled_by?: string | null;
  last_change_reason?: string | null;
  override_enabled: boolean;
  override_reason?: string | null;
};

export type WorkerStatus = {
  shop_id: string;
  status: string;
  process_id?: number | null;
  last_seen_at?: string | null;
  websocket_status: string;
  summary: string;
  ai_enabled: boolean;
  consistency_status: string;
  attention_required: boolean;
  recommended_action?: string | null;
};

export function createOnboardingSession(payload: CreateOnboardingPayload) {
  return apiPost<OnboardingSession>("/api/shops/onboarding", payload);
}

export function getOnboardingSession(sessionId: string) {
  return apiGet<OnboardingSession>(`/api/shops/onboarding/${encodeURIComponent(sessionId)}`);
}

export function submitSmsCode(sessionId: string, smsCode: string) {
  return apiPost<OnboardingSession>(`/api/shops/onboarding/${encodeURIComponent(sessionId)}/submit-sms-code`, {
    sms_code: smsCode
  });
}

export function submitCaptcha(sessionId: string, captchaCode: string) {
  return apiPost<OnboardingSession>(`/api/shops/onboarding/${encodeURIComponent(sessionId)}/submit-captcha`, {
    captcha_code: captchaCode
  });
}

export function cancelOnboardingSession(sessionId: string) {
  return apiPost<OnboardingSession>(`/api/shops/onboarding/${encodeURIComponent(sessionId)}/cancel`, {});
}

export function checkRemoteBrowserLogin(sessionId: string) {
  return apiPost<OnboardingSession>(`/api/shops/onboarding/${encodeURIComponent(sessionId)}/check-login`, {});
}

export function bindShopIdentity(sessionId: string, payload: BindShopIdentityPayload) {
  return apiPost<OnboardingSession>(`/api/shops/onboarding/${encodeURIComponent(sessionId)}/bind-shop-identity`, payload);
}

export function getShopAuthStatus(shopId: string) {
  return apiGet<AuthStatus>(`/api/shops/${encodeURIComponent(shopId)}/auth-status`);
}

export function getOnboardingChecklist(shopId: string) {
  return apiGet<OnboardingChecklist>(`/api/shops/${encodeURIComponent(shopId)}/onboarding-checklist`);
}

export function markNoSendValidationPassed(shopId: string, summary: string, operator = "local_admin") {
  return apiPost<ValidationRun>(`/api/shops/${encodeURIComponent(shopId)}/onboarding-validation/mark-passed`, {
    operator,
    summary
  });
}

export function getAiStatus(shopId: string) {
  return apiGet<AiStatus>(`/api/shops/${encodeURIComponent(shopId)}/ai-status`);
}

export function enableAi(shopId: string, payload: { operator?: string; confirm: boolean; override?: boolean; override_reason?: string | null }) {
  return apiPost<AiStatus>(`/api/shops/${encodeURIComponent(shopId)}/enable-ai`, payload);
}

export function disableAi(shopId: string, reason: string, operator = "local_admin") {
  return apiPost<AiStatus>(`/api/shops/${encodeURIComponent(shopId)}/disable-ai`, { operator, reason });
}

export function getWorkerStatus(shopId: string) {
  return apiGet<WorkerStatus>(`/api/shops/${encodeURIComponent(shopId)}/worker-status`);
}
